Evermeet theme

tessus opened the pull request:
I've created 2 theme files, because on low resolution screens, unicode characters might overlap with the following character. Thus there has to be an extra space after the unicode character.

For high res (retina) screens (which are more and more the standard these days) an extra space looks very unprofessional.

ogr3 commented:
Hi,

Tried the theme, and it worked well.

A note about code standard:

You should use the colors defined in prompt-colors.sh - makes the code easier to read.

You don't have to override variables defined in Default.bgptheme that you don't change (i.e `GIT_PROMPT_SEPARATOR`.

The naming standard used in the other themes for low-res themes are the suffix `_Ubuntu` (since Ubuntus terminal font is one of the reasons you need the extra space after some characters)

So, `Evergreen` and `Evergreen_Ubuntu` would be my choice for naming.

tessus commented:
Thank you for your reply.

It's not `Evergreen`, but `Evermeet`. But I will change the `lowres_screens` to `Ubuntu`. 

I'll also look into using color names. You are right, it is easier to read, but I created that prompt with my own color definitions and the escape sequences were the result. I haven't thought of going back and use the project's color names. But I can do that.

With regards to overriding variables: usually I agree, but I have learned years ago that explicitly defining values can solve many, many problems. What happens, if the default ever changes? Believe me, I've seen this more often than you might think. When I think of it, I have even missed a few that I should've explicitly set - even though they are the same as the default. So if you don't mind, I'd rather add a few variables that are the current default. 

I'm gonna make te changes tomorrow.

ogr3 commented:
Sorry about the name mixup.

You can checkout the other `_Ubuntu` suffixed themes to see which characters we have had to add extra spaces to.

tessus commented:
I've replaced the escape sequences with color names and renamed `Evermeet_lowres_screens.bgptheme` to `Evermeet_Ubuntu.bgptheme`.
I think this pull request is ready to be merged.

ogr3 commented on themes/Evermeet_Ubuntu.bgptheme:
```diff
@@ -0,0 +1,31 @@
+	GIT_PROMPT_SEPARATOR="|"              # separates each item
+
+	GIT_PROMPT_BRANCH="${Magenta}"        # the git branch that is active in the current directory
+	GIT_PROMPT_STAGED="${Cyan}●"          # the number of staged files/directories
```
This one needs and extra space in Ubuntu

ogr3 commented:
The GIT_PROMPT_STAGED circle needs a training space in `Evermeet_Ubuntu`.

ogr3 commented on themes/Evermeet_Ubuntu.bgptheme:
```diff
@@ -0,0 +1,31 @@
+	GIT_PROMPT_CHANGED="${Cyan}✚ "        # the number of changed files
+
+	GIT_PROMPT_UNTRACKED="${Cyan}…"       # the number of untracked files/dirs
+	GIT_PROMPT_STASHED="${Magenta}⚑"      # the number of stashed files/dir
```
This one needs an extra space in Ubuntu

ogr3 commented on themes/Evermeet_Ubuntu.bgptheme:
```diff
@@ -0,0 +1,31 @@
+
+	GIT_PROMPT_UNTRACKED="${Cyan}…"       # the number of untracked files/dirs
+	GIT_PROMPT_STASHED="${Magenta}⚑"      # the number of stashed files/dir
+	GIT_PROMPT_CLEAN="${BoldGreen}✔"      # a colored flag indicating a "clean" repo
```
This one needs an extra space in Ubuntu

tessus commented:
@ogr3 You haven't specified line numbers. _This one needs an extra space_ doesn't really help.

But it seems that a prompt for lowres screens and Ubuntu are different after all.

If I don't add the space to `GIT_PROMPT_STAGED` it looks like this (nice):
![prompt1](https://cloud.githubusercontent.com/assets/223439/15896136/ea8fbf2c-2d5c-11e6-9b78-c9f32561c5ac.png)
When I add the space, it looks like this (not nice):
![promp2](https://cloud.githubusercontent.com/assets/223439/15896154/018e4f90-2d5d-11e6-87be-bb26461f2619.png)

So how do you want to handle this? Shall I create another one that works on my MacBook 2008 lowres (non-retina) screen?

ogr3 commented:
Sorry, I added a line note in github, but I see that the actual lines might not be easily readable.

The lines that need extra space in Ubuntu are:

```
GIT_PROMPT_STAGED="${Cyan}●"
GIT_PROMPT_STASHED="${Magenta}⚑" 
GIT_PROMPT_CLEAN="${BoldGreen}✔" 
```

If there are differences between Ubuntu and mac low res we could just as well add the suffix `_Lowres` for that one.

tessus commented:
No worries. Usually highlighted lines are yellow, but I don't see any. Maybe something wrong with gh.

Yes, these lines also have unicode characters, but for some reason I don't need the extra space for these 3 on my non-retina MacBook screen. The problem with the extra space (if not needed) is that it looks even worse when you connect to that machine on a high res display.

I'll add the spaces and create a _Lowres as well. The file is not big anyway.

tessus commented:
@ogr3 everything in order?

ogr3 commented:
I tried the Ubuntu variant on Ubuntu 16.04 and it works fine.

Looks good to me.

magicmonty commented:
Thanks a lot for the har work.
